### PR TITLE
Replace env var usage with Flyway configuration

### PR DESCRIPTION
**Pull request by jdpgrailsdev**
## What
* Ensure that migration scripts use the same configuration as the Gradle tasks

## How
* Remove usage of environment variables from migration script/replace with use of Flyway provided configuration for database connectivity.

## Recommended reading order
1. `V0_30_22_001__Store_last_sync_state.java`

### CODE REVIEW

**Review by terencecho — DISMISSED** [review]

**Review by terencecho — COMMENTED** [review]
still running into an error when running `./gradlew :airbyte-db:db-lib:dumpConfigsSchema`. but i no longer see the warning about missing env vars.

File: airbyte-db/db-lib/src/main/java/io/airbyte/db/instance/configs/migrations/V0_30_22_001__Store_last_sync_state.java [hunk 1]
```diff
@@ -48,32 +47,13 @@ public class V0_30_22_001__Store_last_sync_state extends BaseJavaMigration {
...
-    this.databaseUrl = databaseUrl;
-  }
-
   @Override
   public void migrate(final Context context) throws Exception {
     LOGGER.info("Running migration: {}", this.getClass().getSimpleName());
     final DSLContext ctx = DSL.using(context.getConnection());
 
-    final Optional<Database> jobsDatabase = getJobsDatabase(databaseUser, databasePassword, databaseUrl);
+    final Optional<Database> jobsDatabase = getJobsDatabase(context.getConfiguration().getUser(),
```
Comment (gosusnp) [inline airbyte-db/db-lib/src/main/java/io/airbyte/db/instance/configs/migrations/V0_30_22_001__Store_last_sync_state.java:55]: This is a weird dependency here between jobs and configs, I am actually wondering if this is still relevant?

**Review by gosusnp — APPROVED** [review]
The change looks reasonable, but I am not sure to which extent it solves our actual problem? Are we dealing a with a config issue or dependency issue?

File: airbyte-db/db-lib/src/main/java/io/airbyte/db/instance/configs/migrations/V0_30_22_001__Store_last_sync_state.java (hunk 1 above)
Comment (jdpgrailsdev) [reply airbyte-db/db-lib/src/main/java/io/airbyte/db/instance/configs/migrations/V0_30_22_001__Store_last_sync_state.java:55]: @gosusnp Agreed. There are a lot of smells in this particular script that relate to ordering of migration.

**Comment by jdpgrailsdev** [comment]
> still running into an error when running `./gradlew :airbyte-db:db-lib:dumpConfigsSchema`. but i no longer see the warning about missing env vars.

@terencecho Could you run with `--stacktrace` and post the output?  I'm curious what error you are now seeing.

**Comment by jdpgrailsdev** [comment]
> The change looks reasonable, but I am not sure to which extent it solves our actual problem? Are we dealing a with a config issue or dependency issue?

@gosusnp My guess is that there are multiple problems, but this one is the first/most obvious.  Fixing the broken windows should lead us to the proper fixes.

**Review by terencecho — APPROVED** [review]

**Pull request merged**